Cargo.toml

```
@@ -17,3 +17,77 @@ exclude = [
   "vendor/web-tree-sitter",
   "vendor/tree-sitter-facade",
 ]
+
+[workspace.dependencies]
+ai_builtins = { git = "https://github.com/getgrit/ai_builtins.git" }
+anyhow = "1.0.70"
+assert_cmd = "2.0.12"
+base64 = "0.21.2"
+buildkite-test-collector = "0.1.1"
+cc = "1.0"
+chrono = "0.4.26"
+clap = "4.1.13"
+clap-markdown = { git = "https://github.com/getgrit/clap-markdown" }
+cli_server = { git = "https://github.com/getgrit/cli_server.git" }
+colored = "2.0.4"
+console = "0.15.7"
+console_error_panic_hook = "0.1.7"
+dashmap = "5.5.3"
+derive_builder = "0.13.1"
+dialoguer = "0.10.4"
+elsa = "1.9.0"
+embeddings = { git = "https://github.com/getgrit/embeddings.git" }
+enum_dispatch = "0.3.12"
+env_logger = "0.10.0"
+extism = "1.0.0-rc7"
+flate2 = { version = "1.0.17", default-features = false }
+fs_extra = "1.3"
+futures = "0.3.29"
+getrandom = "0.2.11"
+git2 = { version = "0.17.2", default-features = false, features = [
+  "vendored-openssl",
+] }
+homedir = "0.2.1"
+http = "0.2.11"
+ignore = "0.4.21"
+im = "15.1.0"
+indicatif = "0.17.5"
+indicatif-log-bridge = "0.2.1"
+insta = "1.30.0"
+itertools = "0.10.5"
+lazy_static = "1.4.0"
+log = "0.4.20"
+markdown = "1.0.0-alpha.11"
+ntest = "0.9.0"
+openssl = "0.10"
+opentelemetry = "0.21.0"
+opentelemetry-otlp = "0.14.0"
+opentelemetry_sdk = "0.21.1"
+path-absolutize = "3.1.1"
+predicates = "3.0.3"
+rand = "0.8.5"
+rayon = "1.8.0"
+regex = "1.7.3"
+reqwest = "0.11.22"
+# Do *NOT* upgrade beyond 1.0.171 until https://github.com/serde-rs/serde/issues/2538 is fixed
+serde = "1.0.164"
```

```suggestion
serde = "<1.0.171"
```

Given the comment, you probably want this version requirement.
"1.0.164" means "1.0.164 or above if semver compatible".